### PR TITLE
feat: Sunset tsms and j2

### PR DESCRIPTION
**Pull request by amirai21**
Removing all TSMs, custom-models, datasets and J2 support.

### CODE REVIEW

File: tests/integration_tests/test-file-9321c0b7-6671-47d2-a96c-044c469cb489.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+<your content goes here>
```
Comment (miri-bar) [inline tests/integration_tests/test-file-9321c0b7-6671-47d2-a96c-044c469cb489.txt:1]: Is this file needed?

File: tests/integration_tests/test-file-d0e9915d-4abd-496b-8330-74ea061822a2.txt [hunk 2]
```diff
@@ -0,0 +1 @@
+<your content goes here>
```
Comment (miri-bar) [inline tests/integration_tests/test-file-d0e9915d-4abd-496b-8330-74ea061822a2.txt:1]: Is this file needed?

File: ai21/__init__.py [hunk 3]
```diff
@@ -100,9 +81,7 @@ def __getattr__(name: str) -> Any:
     "ModelPackageDoesntExistError",
     "TooManyRequestsError",
     "AI21BedrockClient",
-    "AI21SageMakerClient",
     "BedrockModelID",
-    "SageMaker",
```
Comment (Josephasafg) [inline ai21/__init__.py:105]: Please remove - `AsyncAI21SageMakerClient`

File: ai21/models/__init__.py [hunk 4]
```diff
@@ -15,30 +12,16 @@
...
 from ai21.models.responses.file_response import FileResponse
-from ai21.models.responses.gec_response import GECResponse, Correction, CorrectionType
-from ai21.models.responses.improvement_response import ImprovementsResponse, Improvement
-from ai21.models.responses.library_answer_response import LibraryAnswerResponse, SourceDocument
-from ai21.models.responses.library_search_response import LibrarySearchResponse, LibrarySearchResult
-from ai21.models.responses.paraphrase_response import ParaphraseResponse, Suggestion
-from ai21.models.responses.segmentation_response import SegmentationResponse
-from ai21.models.responses.summarize_by_segment_response import SummarizeBySegmentResponse, SegmentSummary, Highlight
-from ai21.models.responses.summarize_response import SummarizeResponse
 from ai21.models.summary_method import SummaryMethod
```
Comment (Josephasafg) [inline ai21/models/__init__.py:16]: What about this?

File: tests/unittests/clients/studio/resources/conftest.py [hunk 5]
```diff
@@ -72,25 +48,6 @@ def mock_async_successful_httpx_response(mocker: MockerFixture) -> httpx.Respons
...
-        {
-            "context": _DUMMY_CONTEXT,
-            "question": _DUMMY_QUESTION,
-        },
-        httpx.Response(status_code=200, json=json_response),
-        _from_dict(obj=AnswerResponse, obj_dict=json_response),
-    )
-
-
 def get_studio_chat(is_async: bool = False):
```
Comment (Josephasafg) [inline tests/unittests/clients/studio/resources/conftest.py:51]: Please Remove

File: tests/unittests/clients/studio/resources/test_async_studio_resource.py [hunk 6]
```diff
@@ -34,18 +20,10 @@ class TestAsyncStudioResources:
     @pytest.mark.asyncio
     @pytest.mark.parametrize(
         ids=[
-            "async_studio_answer",
             "async_studio_chat",
```
Comment (Josephasafg) [inline tests/unittests/clients/studio/resources/test_async_studio_resource.py:23]: Remove

File: tests/unittests/clients/studio/resources/test_studio_resources.py [hunk 7]
```diff
@@ -1,49 +1,26 @@
...
-import httpx
 from ai21.http_client.http_client import AI21HTTPClient
-from ai21.clients.studio.resources.studio_answer import StudioAnswer
 from ai21.clients.studio.resources.studio_resource import StudioResource
-from ai21.models import AnswerResponse
-from ai21.models._pydantic_compatibility import _to_dict
 from ai21.models.ai21_base_model import AI21BaseModel
 from tests.unittests.clients.studio.resources.conftest import (
-    get_studio_answer,
     get_studio_chat,
```
Comment (Josephasafg) [inline tests/unittests/clients/studio/resources/test_studio_resources.py:8]: Remove

**Review by Josephasafg — APPROVED** [review]
